### ISSUE TITLE
Usage of loss which is not CrossEntropy

### ISSUE THREAD

**Issue by dedeswim** [nimarb/pytorch_influence_functions#14]
Hi,

Thank you very much for this implementation!

I noticed that the only allowed loss is `CrossEntropyLoss`. As I'd need your module to compute Influence Functions on other kinds of losses, how about adding the possibility to use a different loss, perhaps passed as param in the `config` dict?

I was thinking about the following changes (to avoid disruptive changes to the API):

https://github.com/nimarb/pytorch_influence_functions/blob/4df5d2ec1baae38d70345740b7eca7466e3b48ef/pytorch_influence_functions/influence_function.py#L8-L9

to

```python
def s_test(z_test, t_test, model, z_loader, gpu=-1, damp=0.01, scale=25.0,
           recursion_depth=5000, loss_fn=compute_loss):
```
And then change
https://github.com/nimarb/pytorch_influence_functions/blob/4df5d2ec1baae38d70345740b7eca7466e3b48ef/pytorch_influence_functions/influence_function.py#L45

to
```python
loss = loss_fn(y, t)
```
With something analogous for:
https://github.com/nimarb/pytorch_influence_functions/blob/4df5d2ec1baae38d70345740b7eca7466e3b48ef/pytorch_influence_functions/influence_function.py#L75

It would be obviously necessary to propagate the `loss_fn` params to the functions that call `grad_z` and `s_test`.

I'd be glad to create a new branch and open a pull request with this change!

Thanks :)
